File: apps/silver_brain_core/lib/item.ex
defmodule SilverBrain.Core.Item do
  use TypedStruct

  @type id() :: String.t()

  typedstruct do
    field :id, id()
    field :name, String.t()
    field :content_type, String.t()
    field :content, String.t()
    field :create_time, String.t()
    field :update_time, String.t()
    field :properties, %{String.t() => String.t()}
    field :parents, [id()]
    field :children, [id()]
  end
end

File: modules/core/src/main/scala/domain/Item.scala
package silverbrain.core

import java.time.Instant

case class Item(
    id: ItemId,
    name: String,
    contentType: Option[String] = None,
    content: Option[String] = None,
    properties: Option[Seq[ItemProperty]] = None,
    parents: Option[Seq[ItemCore]] = None,
    children: Option[Seq[ItemCore]] = None,
    createTime: Option[Instant] = None,
    updateTime: Option[Instant] = None
):
  def toItemCore(): ItemCore = ItemCore(id, name)
